src/schnetpack/atomistic/model.py

```diff
@@ -95,10 +95,13 @@ def __init__(
         self.output_modules = nn.ModuleList(output_modules)
         self.pp = postprocess or []
 
-        self.required_derivatives = set()
+        self.required_derivatives = []
         for m in self.output_modules:
             if hasattr(m, "required_derivatives"):
-                self.required_derivatives.update(m.required_derivatives)
+                for p in m.required_derivatives:
+                    if p not in self.required_derivatives:
+                        self.required_derivatives.append(p)
+
```

Comment:
couldn't you keep it a set and then do: `self.required_derivatives=list(required_derivatives)`?

Reply:
This was a leftover from the torch script trial. Reverted it to original set version